Framework/Algorithms/test/EditInstrumentGeometryTest.h

```diff
@@ -58,14 +53,11 @@ class EditInstrumentGeometryTest : public CxxTest::TestSuite {
         workspace = boost::dynamic_pointer_cast<Mantid::API::MatrixWorkspace>(
             Mantid::API::AnalysisDataService::Instance().retrieve("inputWS")));
 
-    Geometry::Instrument_const_sptr instrument = workspace->getInstrument();
+    const auto &spectrumInfo = workspace->spectrumInfo();
+    TS_ASSERT_EQUALS(spectrumInfo.hasUniqueDetector(0), true);
 
-    auto detids = workspace->getSpectrum(0).getDetectorIDs();
-    TS_ASSERT_EQUALS(detids.size(), 1);
-    detid_t detid = *detids.rbegin();
-    Geometry::IDetector_const_sptr detector = instrument->getDetector(detid);
```

Comment:
Nice :+1: -- I was about to write that you have to use `DetectorInfo`, since the original code used `Instrument::getDetector` (instead of `MatrixWorkspace::getDetector`), but the way you solved it is much nicer (thanks to the unique ID).